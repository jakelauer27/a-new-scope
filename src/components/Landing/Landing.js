import React from 'react'
import propTypes from 'prop-types'

export default function Landing({ continueToSite, episode }) {
  return (
    <div className='landing'>
      <header className='landing-header'>
        <div className='header-right'>
          <h2 className='landing-title'>{episode.title}</h2>
          <h2 className='landing-year'>{episode.year.substring(0, 4)}</h2>
        </div>
        <div className='continue-to-site-btn' onClick={() => continueToSite('menu')}>
          <img src='./images/outerring.svg' alt='ring' className='outerring ring'></img>
          <img src='./images/outerring.svg' alt='ring' className='middlering ring'></img>
          <img src='./images/outerring.svg' alt='ring' className='innerring ring'></img>
          <h2 class='enter-btn'>ENTER</h2>
        </div>
        <img className='starwars-logo' alt='Star Wars Logo' src='./images/starwars_logo.svg'></img>
      </header>
      <section className="scroll-text">
        <div className='fade-overlay'></div>
        <div className="crawl">
          <div className="title">
            <h1 className='scroll-title'>{episode.title}</h1>
          </div>      
          <p>{episode.text}</p>     
        </div>
      </section>
      <img className='millenium-landing' alt='Millenium Falcon' src='./images/millenium_color.png'></img>
    </div>
  )
}

/***************Private

*****************/

Landing.propTypes = {
  continueToSite: propTypes.func.isRequired,

}