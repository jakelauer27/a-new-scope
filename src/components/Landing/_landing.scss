@font-face {
  font-family: 'Jedi';
  src: url('../../utils/fonts/Starjedi.ttf') format('truetype');
}

@font-face {
  font-family: 'Pathway Gothic One';
  src: url('../../utils/fonts/PathwayGothicOne-Regular.ttf') format('truetype');
}

body {
  background-color: black;
}

.main-title2 {
  color: #FCDD2B;
  font-family: 'Jedi';
  position: relative;
  font-size: 3rem;
  letter-spacing: 2px;
  line-height: 100%;
  top: -15px;
  right: 5px;
  height: 50px;
  width: 200px;
  opacity: 0;
  animation: fadeIn 3s ease 4.2s;
  animation-fill-mode: both;
}

.landing {
  color: #FCDD2B;
  font-family: 'Roboto', sans-serif;
  height: 100vh;
  overflow: hidden;
  width: 100%;
}

.landing-header {
  display: flex;
  justify-content: space-between;
  padding: 10px 15px 15px 30px;
  height: 300px;
}

.header-right {
  width: 200px;
}

.landing-title {
  animation: fadeIn 3s ease 4.2s;
  animation-fill-mode: both;
}

.landing-year {
  animation: fadeIn 3s ease 4.2s;
  animation-fill-mode: both;
  font-weight: 300;
  font-style: italic;
  margin-top: -10px;
}

.starwars-logo {
  animation: fadeIn 3s ease 4.2s;
  animation-fill-mode: both;
  height: 89px;
  width: 200px;
}

.continue-to-site-btn {
  animation: fadeIn 3s ease 4.2s;
  animation-fill-mode: both;
  position: relative;
  flex-grow: 1;
  display: flex;
  justify-content: center;
  z-index: 5;
}

.enter-btn {
  color: white;
  font-family: 'Pathway Gothic One', sans-serif;
  font-size: 3em;
  position: relative;
  top: 240px;
  text-shadow: 1px 1px 1px white;
  height: 80px;
  transition: all .4s;
  z-index: 100;

  &:hover {
    color: #00D1FF;
    text-shadow: 1px 1px 1px #00D1FF;
    cursor: pointer;
  }

  &:hover ~ .ring {
    animation: rotateToTop .5s ease;
    animation-fill-mode: forwards;
    cursor: pointer;
  }
}

.ring {
  position: absolute;
  transition: all .5s;
}

.outerring {
  animation: rotateRing 20s linear 0s infinite;
  height: 280px;
  width: 280px;
}

.middlering {
  animation: rotateRingMiddle 23s linear 0s infinite;
  top: 40px;
  height: 200px;
  transform: rotate(170deg);
  width: 200px;
}

.innerring {
  animation: rotateRingInner 29s linear 0s infinite;
  top: 90px;
  height: 100px;
  transform: rotate(70deg);
  width: 100px;
}

.scroll-text{
  display: flex;
  justify-content: center;
  height: 80%;
  perspective: 400px;
  color: #feda4a;
  font-family: 'Pathway Gothic One', sans-serif;
  font-size: 500%;
  font-weight: 600;
  letter-spacing: 6px;
  line-height: 150%;
  position: relative;
  overflow: hidden;
  text-align: justify;
}

.scroll-title {
  text-align: center;
}

.fade-overlay {
  height: 550px;
  position: absolute;
  top: -20px;
  background-image: linear-gradient(black 2%, rgba(0, 0, 0, 0) 98%);
  width: 100%;
  z-index: 1;
}

.crawl {
  animation: crawl 80s linear;
  animation-fill-mode: forwards;
  position: relative;
  top: 0px;
  transform-origin: 50% 100%;
}

.millenium-landing {
  animation: milleniumZoom 5s ease;
  animation-fill-mode: forwards;
  position: absolute;
  left: calc(50% - 300px);
  transform: perspective(400px);
  bottom: 160px;
  transform: rotate(-60deg) perspective(50em) rotateX(38deg) rotateY(55deg);
}

@keyframes fadeIn {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}

@keyframes rotateRing {
  100% {
    transform: rotate(360deg);
  }
}

@keyframes rotateRingMiddle {
  100% {
    transform: rotate(530deg);
  }
}

@keyframes rotateRingInner {
  100% {
    transform: rotate(430deg);
  }
}



@keyframes rotateToTop {
  100% {
    transform: rotate(50deg);
  }
}

@keyframes milleniumZoom {
  0% {
    transform: 
      translateX(-500px) 
      translateY(560px)
  }
  
  60% {
    transform: 
      rotate(-65deg) 
      perspective(50em) 
      rotateX(38deg) 
      rotateY(55deg) 
      perspective(50em) 
      translate(-60px, 40px)
      scale(1);
  }

  87% {
    transform: 
      rotate(-70deg) 
      perspective(50em) 
      rotateX(38deg) 
      rotateY(55deg) 
      perspective(50em) 
      translate(-44px, 45px)
      scale(1.1);
  }
  99% {
    transform: 
      rotate(-80deg) 
      perspective(50em) 
      rotateX(38deg) 
      rotateY(55deg) 
      translate(1034px, -1000px)
      scale(0.1)
  }
  100% {
    transform: 
      rotate(-800deg) 
      perspective(50em) 
      rotateX(38deg) 
      rotateY(55deg) 
      translate(1034px, -1000px)
      scale(0)
      
  }
}

@keyframes crawl {
  0% {
    top: 100%;
    transform: rotateX(20deg) translateZ(0);
  }
  100% { 
    top: -6000px;
    transform: rotateX(25deg) translateZ(-2500px);
  }
}