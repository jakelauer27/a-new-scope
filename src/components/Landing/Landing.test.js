import React from 'react'
import { shallow } from 'enzyme'
import Landing from './Landing'

describe('Landing', () => {
  const mockChange = jest.fn()
  const mockepisode = { title: 'A New Hope', year: '1995', text: 'text' }
  const wrapper = shallow(
    <Landing
      continueToSite={mockChange}
      episode={mockepisode}
    />
  )

  it('should change to the menu screen when the enter button is clicked', () => {
    wrapper.find('.enter-btn').simulate('click')
    expect(mockChange).toBeCalled()
    expect(mockChange).toBeCalledWith('menu')
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
