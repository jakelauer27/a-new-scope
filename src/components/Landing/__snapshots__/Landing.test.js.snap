// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Landing
    continueToSite={
      [MockFunction] {
        "calls": Array [
          Array [
            "menu",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    episode={
      Object {
        "text": "text",
        "title": "A New Hope",
        "year": "1995",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="landing-header"
        >
          <div
            className="header-right"
          >
            <h2
              className="landing-title"
            >
              A New Hope
            </h2>
            <h2
              className="landing-year"
            >
              1995
            </h2>
          </div>
          <div
            className="continue-to-site-btn"
            onClick={[Function]}
          >
            <img
              alt="ring"
              className="outerring ring"
              src="./images/outerring.svg"
            />
            <img
              alt="ring"
              className="middlering ring"
              src="./images/outerring.svg"
            />
            <img
              alt="ring"
              className="innerring ring"
              src="./images/outerring.svg"
            />
            <h2
              class="enter-btn"
            >
              ENTER
            </h2>
          </div>
          <img
            alt="Star Wars Logo"
            className="starwars-logo"
            src="./images/starwars_logo.svg"
          />
        </header>,
        <section
          className="scroll-text"
        >
          <div
            className="fade-overlay"
          />
          <div
            className="crawl"
          >
            <div
              className="title"
            >
              <h1
                className="scroll-title"
              >
                A New Hope
              </h1>
            </div>
            <p>
              text
            </p>
          </div>
        </section>,
        <img
          alt="Millenium Falcon"
          className="millenium-landing"
          src="./images/millenium_color.png"
        />,
      ],
      "className": "landing",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="header-right"
            >
              <h2
                className="landing-title"
              >
                A New Hope
              </h2>
              <h2
                className="landing-year"
              >
                1995
              </h2>
            </div>,
            <div
              className="continue-to-site-btn"
              onClick={[Function]}
            >
              <img
                alt="ring"
                className="outerring ring"
                src="./images/outerring.svg"
              />
              <img
                alt="ring"
                className="middlering ring"
                src="./images/outerring.svg"
              />
              <img
                alt="ring"
                className="innerring ring"
                src="./images/outerring.svg"
              />
              <h2
                class="enter-btn"
              >
                ENTER
              </h2>
            </div>,
            <img
              alt="Star Wars Logo"
              className="starwars-logo"
              src="./images/starwars_logo.svg"
            />,
          ],
          "className": "landing-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2
                  className="landing-title"
                >
                  A New Hope
                </h2>,
                <h2
                  className="landing-year"
                >
                  1995
                </h2>,
              ],
              "className": "header-right",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "A New Hope",
                  "className": "landing-title",
                },
                "ref": null,
                "rendered": "A New Hope",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "1995",
                  "className": "landing-year",
                },
                "ref": null,
                "rendered": "1995",
                "type": "h2",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="ring"
                  className="outerring ring"
                  src="./images/outerring.svg"
                />,
                <img
                  alt="ring"
                  className="middlering ring"
                  src="./images/outerring.svg"
                />,
                <img
                  alt="ring"
                  className="innerring ring"
                  src="./images/outerring.svg"
                />,
                <h2
                  class="enter-btn"
                >
                  ENTER
                </h2>,
              ],
              "className": "continue-to-site-btn",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "ring",
                  "className": "outerring ring",
                  "src": "./images/outerring.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "ring",
                  "className": "middlering ring",
                  "src": "./images/outerring.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "ring",
                  "className": "innerring ring",
                  "src": "./images/outerring.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "ENTER",
                  "class": "enter-btn",
                },
                "ref": null,
                "rendered": "ENTER",
                "type": "h2",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Star Wars Logo",
              "className": "starwars-logo",
              "src": "./images/starwars_logo.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="fade-overlay"
            />,
            <div
              className="crawl"
            >
              <div
                className="title"
              >
                <h1
                  className="scroll-title"
                >
                  A New Hope
                </h1>
              </div>
              <p>
                text
              </p>
            </div>,
          ],
          "className": "scroll-text",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fade-overlay",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="title"
                >
                  <h1
                    className="scroll-title"
                  >
                    A New Hope
                  </h1>
                </div>,
                <p>
                  text
                </p>,
              ],
              "className": "crawl",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <h1
                    className="scroll-title"
                  >
                    A New Hope
                  </h1>,
                  "className": "title",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "A New Hope",
                    "className": "scroll-title",
                  },
                  "ref": null,
                  "rendered": "A New Hope",
                  "type": "h1",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "text",
                },
                "ref": null,
                "rendered": "text",
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Millenium Falcon",
          "className": "millenium-landing",
          "src": "./images/millenium_color.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="landing-header"
          >
            <div
              className="header-right"
            >
              <h2
                className="landing-title"
              >
                A New Hope
              </h2>
              <h2
                className="landing-year"
              >
                1995
              </h2>
            </div>
            <div
              className="continue-to-site-btn"
              onClick={[Function]}
            >
              <img
                alt="ring"
                className="outerring ring"
                src="./images/outerring.svg"
              />
              <img
                alt="ring"
                className="middlering ring"
                src="./images/outerring.svg"
              />
              <img
                alt="ring"
                className="innerring ring"
                src="./images/outerring.svg"
              />
              <h2
                class="enter-btn"
              >
                ENTER
              </h2>
            </div>
            <img
              alt="Star Wars Logo"
              className="starwars-logo"
              src="./images/starwars_logo.svg"
            />
          </header>,
          <section
            className="scroll-text"
          >
            <div
              className="fade-overlay"
            />
            <div
              className="crawl"
            >
              <div
                className="title"
              >
                <h1
                  className="scroll-title"
                >
                  A New Hope
                </h1>
              </div>
              <p>
                text
              </p>
            </div>
          </section>,
          <img
            alt="Millenium Falcon"
            className="millenium-landing"
            src="./images/millenium_color.png"
          />,
        ],
        "className": "landing",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="header-right"
              >
                <h2
                  className="landing-title"
                >
                  A New Hope
                </h2>
                <h2
                  className="landing-year"
                >
                  1995
                </h2>
              </div>,
              <div
                className="continue-to-site-btn"
                onClick={[Function]}
              >
                <img
                  alt="ring"
                  className="outerring ring"
                  src="./images/outerring.svg"
                />
                <img
                  alt="ring"
                  className="middlering ring"
                  src="./images/outerring.svg"
                />
                <img
                  alt="ring"
                  className="innerring ring"
                  src="./images/outerring.svg"
                />
                <h2
                  class="enter-btn"
                >
                  ENTER
                </h2>
              </div>,
              <img
                alt="Star Wars Logo"
                className="starwars-logo"
                src="./images/starwars_logo.svg"
              />,
            ],
            "className": "landing-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h2
                    className="landing-title"
                  >
                    A New Hope
                  </h2>,
                  <h2
                    className="landing-year"
                  >
                    1995
                  </h2>,
                ],
                "className": "header-right",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "A New Hope",
                    "className": "landing-title",
                  },
                  "ref": null,
                  "rendered": "A New Hope",
                  "type": "h2",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "1995",
                    "className": "landing-year",
                  },
                  "ref": null,
                  "rendered": "1995",
                  "type": "h2",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <img
                    alt="ring"
                    className="outerring ring"
                    src="./images/outerring.svg"
                  />,
                  <img
                    alt="ring"
                    className="middlering ring"
                    src="./images/outerring.svg"
                  />,
                  <img
                    alt="ring"
                    className="innerring ring"
                    src="./images/outerring.svg"
                  />,
                  <h2
                    class="enter-btn"
                  >
                    ENTER
                  </h2>,
                ],
                "className": "continue-to-site-btn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "ring",
                    "className": "outerring ring",
                    "src": "./images/outerring.svg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "ring",
                    "className": "middlering ring",
                    "src": "./images/outerring.svg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "ring",
                    "className": "innerring ring",
                    "src": "./images/outerring.svg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "ENTER",
                    "class": "enter-btn",
                  },
                  "ref": null,
                  "rendered": "ENTER",
                  "type": "h2",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Star Wars Logo",
                "className": "starwars-logo",
                "src": "./images/starwars_logo.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="fade-overlay"
              />,
              <div
                className="crawl"
              >
                <div
                  className="title"
                >
                  <h1
                    className="scroll-title"
                  >
                    A New Hope
                  </h1>
                </div>
                <p>
                  text
                </p>
              </div>,
            ],
            "className": "scroll-text",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fade-overlay",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="title"
                  >
                    <h1
                      className="scroll-title"
                    >
                      A New Hope
                    </h1>
                  </div>,
                  <p>
                    text
                  </p>,
                ],
                "className": "crawl",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <h1
                      className="scroll-title"
                    >
                      A New Hope
                    </h1>,
                    "className": "title",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "A New Hope",
                      "className": "scroll-title",
                    },
                    "ref": null,
                    "rendered": "A New Hope",
                    "type": "h1",
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "text",
                  },
                  "ref": null,
                  "rendered": "text",
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Millenium Falcon",
            "className": "millenium-landing",
            "src": "./images/millenium_color.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
